opentaskspal/src/main/java/org/dmfs/opentaskspal/readdata/DueTime.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.opentaskspal.readdata;
+
+import org.dmfs.android.contentpal.RowDataSnapshot;
+import org.dmfs.opentaskspal.readdata.temp.SinglePresent;
+import org.dmfs.optional.Absent;
+import org.dmfs.optional.Optional;
+import org.dmfs.optional.Present;
+import org.dmfs.optional.decorators.DelegatingOptional;
+import org.dmfs.rfc5545.DateTime;
+import org.dmfs.rfc5545.Duration;
+import org.dmfs.tasks.contract.TaskContract.Tasks;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+
+/**
+ * {@link Optional} for the due {@link DateTime} of a task.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class DueTime extends DelegatingOptional<DateTime>
+{
+    public static final String[] PROJECTION;
+
+    // TODO Either switch to Iterables (maybe at {@link View#withProjection} as well), or create support for array operations
```

Comment:
I already considered a `Projection` type which can be composed from multiple projections. It might even make sense to move the projection from being a `View` property to being a `RowSet` property which then passes it as a parameter to `View.rows(…)`.

Reply:
I can see that that could help a bit, but that wouldn't replace this need to define the required projections as constants on read data classes, right? It's really not nice but since querying and reading the individual columns is independent, I don't see a different solution now. Do you?

About this `Projection` type and possibly changing `View`/`RowSet`, shall we create a ticket in contentpal?
I assume it wouldn't be worth holding this subtask feature up because of that though.

Reply:
FTR: I created this https://github.com/dmfs/ContentPal/issues/134

In general we can't really guarantee that the columns you want to read are present in a `RowDataSnapshot`. Also IMO that's not really the job of the types we have so far.

In order to avoid these static fields you would have to create new types like `Query` and `Entity` which provide the required information to `RowSet` and `View`. IMO that's subject to the next layer on top of ContentPal. It could work like this (sort of):

```java
Iterable<EntitySet<Tasks>> tasks =
    new Query<Tasks>(tasksView,
        new Seq<>(
            new DueEntity(),
            new StartEntity(),
            new TitleEntity())).results();
```

`EntitySet` would have a method to retrieve the value of a specific `Entity`.

The other option would be to always request all fields, that's the only guarantee that the column you need is in the `RowDataSnapshot` you have.

Reply:
How about doing that for now for subtasks loading, to omit selection and loading all columns?
I suppose the query's speed is the same, to find the rows, and since we won't have many subtasks usually the overhead of loading those extra columns into memory is negligible.

I would remove these constants and the projection from the query.